qiskit/pulse/commands/parametric_pulses.py

```diff
@@ -0,0 +1,385 @@
+    """
+    A truncated pulse envelope shaped according to the Gaussian function whose mean is centered at
+    the center of the pulse (duration / 2):
+
+        f(x) = amp * exp( -(1/2) * (x - duration/2)^2 / sigma^2) )  ,  0 <= x < duration
+    """
+
+    def __init__(self,
+                 duration: int,
+                 amp: complex,
+                 sigma: float):
+        """Initialize the gaussian pulse.
+
+        Args:
+            duration: Pulse length in terms of the the sampling period `dt`.
+            amp: The amplitude of the Gaussian envelope.
+            sigma: A measure of how wide or narrow the Gaussian peak is; described mathematically
+                   in the class docstring.
+        """
+        self._amp = complex(amp)
+        self._sigma = sigma
+        super().__init__(duration=duration)
+
+    @property
+    def amp(self):
+        return self._amp
+
+    @property
+    def sigma(self):
+        return self._sigma
+
+    def get_sample_pulse(self) -> SamplePulse:
+        return gaussian(duration=self.duration, amp=self.amp,
+                        sigma=self.sigma, zero_ends=False)
+
+    def validate_parameters(self) -> None:
+        if abs(self.amp) > 1.:
+            raise PulseError("The amplitude norm must be <= 1, "
+                             "found: {}".format(abs(self.amp)))
+        if self.sigma <= 0:
+            raise PulseError("Sigma must be greater than 0.")
+
+    @property
+    def parameters(self) -> Dict[str, Any]:
+        return {"duration": self.duration, "amp": self.amp, "sigma": self.sigma}
+
+    def __repr__(self):
+        return '{}(duration={}, amp={}, sigma={})' \
+               ''.format(self.__class__.__name__, self.duration, self.amp, self.sigma)
+
+
+class GaussianSquare(ParametricPulse):
+    """
+    A square pulse with a Gaussian shaped risefall on either side:
+
+        risefall = (duration - width) / 2
+
+    0 <= x < risefall
+        f(x) = amp * exp( -(1/2) * (x - risefall/2)^2 / sigma^2) )
+
+    risefall <= x < risefall + width
+        f(x) = amp
+
+    risefall + width <= x < duration
+        f(x) = amp * exp( -(1/2) * (x - (risefall + width)/2)^2 / sigma^2) )
+    """
+
+    def __init__(self,
+                 duration: int,
+                 amp: complex,
+                 sigma: float,
+                 width: float):
+        """Initialize the gaussian square pulse.
+
+        Args:
+            duration: Pulse length in terms of the the sampling period `dt`.
+            amp: The amplitude of the Gaussian and of the square pulse.
+            sigma: A measure of how wide or narrow the Gaussian risefall is; see the class
+                   docstring for more details.
+            width: The duration of the embedded square pulse.
+        """
+        self._amp = complex(amp)
+        self._sigma = sigma
+        self._width = width
+        super().__init__(duration=duration)
+
+    @property
+    def amp(self):
+        return self._amp
```

Comment:
why not having `self.amp` directly?

Reply:
To prevent it from being updated by the user